### PR TITLE
Added a new game GoFish

### LINKED ISSUE #4280: [New game]: Go Fish
### 🎮 Game Request

Go Fish is a classic card game , The objective of the game is to collect as many pairs of cards as possible. Players take turns asking each other for specific cards to match the ones they already have in their hand. If the opponent has the requested card, they must give it to the player. If not, the player "goes fish" and draws a card from the deck. The game continues until all pairs have been collected or the deck is empty. The player with the most pairs at the end wins!

### Point down the features

Game Logic:
Creation of a deck of cards (52 cards).
Shuffling of the deck before dealing cards.
Dealing 7 cards to each player at the beginning of the game.
Implementing the rules of the game (asking for cards, drawing from the deck, forming pairs).
Tracking pairs and updating the score.

Computer Player:

AI for the computer player to make decisions.
Strategy for the computer player's turn (e.g., choosing which cards to ask for).

End Game:

Detecting the end of the game (when all pairs are formed or the deck is empty.
Declaring the winner based on the number of pairs collected.

### Select program in which you are contributing

GSSoC24

### Code of Conduct

- [X] I follow [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md) of this project.

### PR DESCRIPTION
**Pull request by pani2004**
Fixes #4280 

 
## Mark the task you have completed ✅

- [ x] I follow [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md) & [CODE OF CONDUCT](https://github.com/kunjgit/GameZone/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [ x] I have performed a self-review of my own code or work.
- [ x] My changes generates no new warnings.
- [ x] I have followed proper naming convention showed in [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md)
- [ x] I have added screenshot for website preview in assets/images 
- [ x] I have added entries for my game in main README.md
- [ x] I have added README.md in my folder 
- [ x] I have specified the respective issue number for which I have requested the new game.

<hr>

## Add your screenshots(Optional) 📸

GoFish

--- 
<br>

## Thank you soo much for contributing to our repository 💗

### CODE REVIEW

**Comment by kunjgit** [comment]
Thank you @pani2004 ,for creating the PR and contributing to our GameZone 💗 
 Review team will review the PR and will reach out to you soon! 😇 
 Make sure that you have marked all the tasks that you are done with ✅. 
 Thank you for your patience! 😀

**Review by kunjgit — CHANGES_REQUESTED** [review]
Hello @pani2004,
You need to make the following changes:
  - Please add a README.md file inside the newly created folder.

Hoping that you will make those changes soon 🚀

**Comment by pani2004** [comment]
@kunjgit Already added the README.md file inside the folder

**Comment by pani2004** [comment]
@ishita-43 Pls review the PR!!!

File: README.md [hunk 1]
```diff
@@ -326,7 +326,7 @@ This repository also provides one such platforms where contributers come over an
 | [Tic-Tac-Toe Game](https://github.com/kunjgit/GameZone/tree/main/Games/Tic-Tac-Toe) |
 | [Rapid_click_frenzy](https://github.com/kunjgit/GameZone/tree/main/Games/Rapid_click_frenzy) 
 |[Penguins Can't Fly](https://github.com/Will2Jacks/GameZoneForked/tree/Task/Games/Penguins_Can't_Fly)|
-
+[GoFish](https://github.com/kunjgit/GameZone/tree/main/Games/GoFish)
```
Comment (ishita-43) [inline README.md:329]: there should be bars as other games had in the ending

File: README.md [hunk 2]
```diff
@@ -326,7 +326,7 @@ This repository also provides one such platforms where contributers come over an
 | [Tic-Tac-Toe Game](https://github.com/kunjgit/GameZone/tree/main/Games/Tic-Tac-Toe) |
 | [Rapid_click_frenzy](https://github.com/kunjgit/GameZone/tree/main/Games/Rapid_click_frenzy) 
 |[Penguins Can't Fly](https://github.com/Will2Jacks/GameZoneForked/tree/Task/Games/Penguins_Can't_Fly)|
```
Comment (ishita-43) [inline README.md:328]: change the ' sign from the name and as well as from the link it is creating merge conflicts

**Review by ishita-43 — CHANGES_REQUESTED** [review]
I hope you will make desired changes soon!!

**Review by kunjgit — CHANGES_REQUESTED** [review]
Hello @pani2004,
You need to make the following changes:
  - Please add a README.md file inside the newly created folder.

Hoping that you will make those changes soon 🚀

**Review by kunjgit — CHANGES_REQUESTED** [review]
Hello @pani2004,
You need to make the following changes:
  - Please add a README.md file inside the newly created folder.

Hoping that you will make those changes soon 🚀

File: README.md (hunk 2 above)
Comment (pani2004) [reply README.md:340]: @ishita-43 I have not added that

**Comment by pani2004** [comment]
@ishita-43 Made the changes

File: README.md (hunk 1 above)
Comment (pani2004) [reply README.md]: @ishita-43  Made the changes

File: README.md (hunk 2 above)
Comment (ishita-43) [reply README.md:340]: i  know but i was asking if you could resolve that .but its okay i will try other way!!

File: README.md [hunk 3]
```diff
@@ -338,7 +338,8 @@ This repository also provides one such platforms where contributers come over an
 | [Brick Buster Game](https://github.com/kunjgit/GameZone/tree/main/Games/Brick Buster) |
 | [Rapid_click_frenzy](https://github.com/kunjgit/GameZone/tree/main/Games/Rapid_click_frenzy) 
 |[Penguins Can't Fly](https://github.com/Will2Jacks/GameZoneForked/tree/Task/Games/Penguins_Can't_Fly)|
-| [Taash_Game](https://github.com/kunjgit/GameZone/tree/main/Games/Taash_Game)|
```
Comment (ishita-43) [inline README.md:341]: add this one again

Comment (pani2004) [reply README.md:341]: @ishita-43 added

**Review by kunjgit — CHANGES_REQUESTED** [review]
Hello @pani2004,
You need to make the following changes:
  - Please add a README.md file inside the newly created folder.

Hoping that you will make those changes soon 🚀

**Comment by pani2004** [comment]
@kunjgit @ishita-43 Pls review this PR and merge

**Review by ishita-43 — APPROVED** [review]
Good to go!!

**Comment by kunjgit** [comment]
Thank you  @pani2004 , for your valuable time and contribution in our GameZone 💗. 
 It’s our GameZone, so Let’s build this GameZone altogether !!🤝
  Hoping to see you soon with another PR again 😇 
 Wishing you all the best for your journey into Open Source🚀

**Pull request merged**